astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -692,6 +692,7 @@ def test_round(self):
         self.check(np.round)
 
     # NUMPY_LT_1_25
+    @pytest.mark.skipif(not NUMPY_LT_2_0, reason="np.round_ is removed in NumPy 2.0")
```

Comment:
I think what is needed here is to only define the function on numpy < 2, i.e.,
```
if NUMPY_LT_2_0:
    @pytest.mark.filterwarnings(...)
    def test_round_(...)
```
The way the test for completeness is done is that defining the test causes the name to be added to a tested list.

Reply:
Hmm, I did not see that fail in https://github.com/astropy/astropy/actions/runs/6052493225/job/16468784807#step:7:1349 or the tests for #15274 or my local tests so far – I'd expect pytest to skip already the function definition here, could this be different behaviour over different pytest versions? In what setup did you see the failure?

Reply:
> The way the test for completeness is done is that defining the test causes the name to be added to a tested list.

Or what did you mean by "test for completeness" – something codecov-related?

Reply:
I'm very confused. When I looked at the tests before, the `-dev` test did have a failure, but now it has all passed. I agree that logically `pytest.skipif` should work, so all is good now!  Let's get it in!!